xcdat/spatial_avg.py

```diff
@@ -338,78 +339,136 @@ def _get_weights(
         }
         axis_weights: AxisWeights = {}
 
-        for dim, bounds_by_type in bounds.items():
-            domain_bounds = bounds_by_type["domain"]
-            region_bounds = bounds_by_type["region"]
-
-            if domain_bounds is not None:
```

Comment:
This check never happens because `self._dataset.bounds.get_bounds()` will raise a ValueError if a domain bounds is not found for the lat/lon axes. I removed this check altogether.